docs/discussion/concepts/overflow.md

```diff
@@ -0,0 +1,208 @@
+overflow too, but it happens far less often in practice.  Even the smallest, `float`, has a range of
+$10^{38}$, while the diameter of the observable universe measured in atomic diameters is "only"
+about $10^{37}$![^1]
+
+[^1]: Here, we take the radius of the observable universe as 46.6 billion light years, and the
+      diameter of a hydrogen atom as 0.1 nanometers.
+
+Of course, many domains prefer the simplicity and interpretability of integral types.  This avoids
+some of the more counterintuitive aspects of floating point arithmetic --- for example, did you know
+that the difference between consecutive representable `double` values can be greater than
+$10^{292}$?  The price we pay for eschewing these complexities is the need to handle overflow, and
+here are the main strategies we've seen for doing so.
+
+### Do nothing
+
+This is the simplest approach, and probably also the most popular: make the users responsible for
+avoiding overflow.  The documentation may simply warn them to check their values ahead of time, as
+in this [example from the bernedom/SI
+library](https://github.com/bernedom/SI/blob/main/doc/implementation-details.md#implicit-ratio-conversion--possible-loss-of-precision).
+
+While this approach is perfectly valid, it does put a lot of responsibility onto the end users, many
+of whom may not realize that they have incurred it.  Even for those who do, we've seen above that
+many unit conversions are hard to spot.  It's reasonable to assume that this approach leads to the
+highest incidence of overflow bugs.
+
+### Curate user-facing types
+
+The [`std::chrono`](https://en.cppreference.com/w/cpp/chrono/duration) library, a time-only units
+library, takes a different approach.  It uses intimate knowledge of the domain to craft its
+user-facing types such that they all cover the same (very generous) range of values. Specifically,
+every `std::chrono::duration` type shorter than a day --- everything from `std::chrono::hours`, all
+the way down to `std::chrono::nanoseconds` --- is guaranteed to be able to represent _at least_ ±292
+years.
+
+As long as users' durations are within this range --- _and_, as long as they _stick to these primary
+user-facing types_ --- they can be confident that their values won't overflow.
+
+This approach works very well in practice for the (great many) users who can meet both of these
+conditions.  However, it doesn't translate well to a _multi-dimensional_ units library: since there
+are many dimensions, and new ones can be created on the fly, it's infeasible to try to define
+a "practical range" for _all_ of them.  Besides, users can still form arbitrary
+`std::chrono::duration` types, and they may not realize the safety they have given up in doing so.
+
+### Adapt to risk
+
+Fundamentally, there are two contributions to the level of overflow risk:
+
+1. The _size of the conversion factor_: **bigger factors** mean **more risk**.
```

Comment:
In the example you gave earlier, it's not the magnitude of the conversion factor between meters and yards that is the problem (indeed that conversion factor doesn't even exist in the integral domain) but the conversion factor to the implied common unit, or the magnitudes of the numerator and denominator of the conversion factor. Not sure whether this needs to be called out explicitly, but I think you have segued somewhat from the original motivating example.

Reply:
Good point.  I ended up going for a footnote here.  I considered discussing the kind of overflow that can happen with rational conversion factors applied to integral-rep quantities.  However, in the context of _this section_, we're talking about conversions with safety checks applied... and rational conversion factors would already fail the safety check for truncation, so they're not really in scope here.